common/decklist.cpp

```diff
@@ -726,12 +726,29 @@ bool DeckList::deleteNode(AbstractDecklistNode *node, InnerDecklistNode *rootNod
 void DeckList::updateDeckHash()
 {
     QStringList cardList;
-    for (int i = 0; i < root->size(); i++) {
+    bool isValidDeckList = true;
+    QSet<QString> hashZones, optionalZones;
```

Comment:
This is what I meant but you can make it more efficient by declaring them statically since they will never change; as written we will create a new QSet object every time this function is called (admittedly, this is called rarely, once, on deck updates.)

In this context it doesn't matter that it's a little less efficient; it should never lead to a user noticing it.

If it's easy to move to a static section though, I'd do that.